2008-07-04  Lukasz Kosewski <lkosewsk@gmail.com>

	* See the comment below.  Mingw is stupid, so we replace instances of 'putc' with 'fputc' when building for WIN32, and everything is happy. 

--- argp.old/argp-fmtstream.h	2008-07-04 20:24:27.000000000 -0400
+++ argp/argp-fmtstream.h	2008-07-04 20:26:16.000000000 -0400
@@ -42,7 +42,15 @@
 /* Disable stdio locking */
 # define FLOCKFILE(f)
 # define FUNLOCKFILE(f)
+#ifndef _WIN32
 # define PUTC_UNLOCKED(c, f) putc((c), (f))
+#else
+/* Mingw does something really dumb with putc (probably evaluating the 'stream'
+ * argument to it, which putc can technically do by the spec), which causes it
+ * to fatally abort.  If we use fputc, this problem goes away
+ */
+# define PUTC_UNLOCKED(c, f) fputc((c), (f))
+#endif
 # define FPUTS_UNLOCKED(s, f) fputs((s), (f))
 # define FWRITE_UNLOCKED(b, s, n, f) fwrite((b), (s), (n), (f))
 #endif /* No thread safe i/o */
